From: Jakub Jirutka <jakub@jirutka.cz>
Date: Wed, 19 Sep 2016 20:14:00 +0200
Subject: [PATCH] Move some unnecessary gems to groups (i.e. exclude them)

diff --git a/Gemfile b/Gemfile
--- a/Gemfile
+++ b/Gemfile
@@ -22,19 +22,19 @@
 gem 'devise',                 '~> 4.0'
 gem 'doorkeeper',             '~> 4.2.0'
 gem 'omniauth',               '~> 1.3.1'
-gem 'omniauth-auth0',         '~> 1.4.1'
-gem 'omniauth-azure-oauth2',  '~> 0.0.6'
+gem 'omniauth-auth0',         '~> 1.4.1', group: :auth_ext
+gem 'omniauth-azure-oauth2',  '~> 0.0.6', group: :auth_ext
 gem 'omniauth-bitbucket',     '~> 0.0.2'
-gem 'omniauth-cas3',          '~> 1.1.2'
-gem 'omniauth-facebook',      '~> 3.0.0'
+gem 'omniauth-cas3',          '~> 1.1.2', group: :auth_ext
+gem 'omniauth-facebook',      '~> 3.0.0', group: :auth_ext
 gem 'omniauth-github',        '~> 1.1.1'
 gem 'omniauth-gitlab',        '~> 1.0.0'
-gem 'omniauth-google-oauth2', '~> 0.4.1'
+gem 'omniauth-google-oauth2', '~> 0.4.1', group: :auth_ext
 gem 'omniauth-kerberos',      '~> 0.3.0', group: :kerberos
-gem 'omniauth-saml',          '~> 1.6.0'
-gem 'omniauth-shibboleth',    '~> 1.2.0'
-gem 'omniauth-twitter',       '~> 1.2.0'
-gem 'omniauth_crowd',         '~> 2.2.0'
+gem 'omniauth-saml',          '~> 1.6.0', group: :auth_ext
+gem 'omniauth-shibboleth',    '~> 1.2.0', group: :auth_ext
+gem 'omniauth-twitter',       '~> 1.2.0', group: :auth_ext
+gem 'omniauth_crowd',         '~> 2.2.0', group: :auth_ext
 gem 'rack-oauth2',            '~> 1.2.1'
 gem 'jwt'
 
@@ -83,19 +83,19 @@
 gem 'carrierwave', '~> 0.10.0'
 
 # Drag and Drop UI
-gem 'dropzonejs-rails', '~> 0.7.1'
+gem 'dropzonejs-rails', '~> 0.7.1', group: :assets
 
 # for backups
-gem 'fog-aws', '~> 0.9'
-gem 'fog-azure', '~> 0.0'
-gem 'fog-core', '~> 1.40'
-gem 'fog-local', '~> 0.3'
-gem 'fog-google', '~> 0.3'
-gem 'fog-openstack', '~> 0.1'
-gem 'fog-rackspace', '~> 0.1.1'
+gem 'fog-aws', '~> 0.9', group: :fog
+gem 'fog-azure', '~> 0.0', group: :fog
+gem 'fog-core', '~> 1.40', group: :fog
+gem 'fog-local', '~> 0.3', group: :fog
+gem 'fog-google', '~> 0.3', group: :fog
+gem 'fog-openstack', '~> 0.1', group: :fog
+gem 'fog-rackspace', '~> 0.1.1', group: :fog
 
 # for aws storage
-gem 'unf', '~> 0.1.4'
+gem 'unf', '~> 0.1.4', group: :fog
 
 # Authorization
 gem 'six', '~> 0.2.0'
@@ -182,7 +182,7 @@
 gem 'ruby-fogbugz', '~> 0.2.1'
 
 # d3
-gem 'd3_rails', '~> 3.5.0'
+gem 'd3_rails', '~> 3.5.0', group: :assets
 
 # underscore-rails
 gem 'underscore-rails', '~> 1.8.0'
@@ -201,10 +201,10 @@
 gem 'rack-attack', '~> 4.3.1'
 
 # Ace editor
-gem 'ace-rails-ap', '~> 4.1.0'
+gem 'ace-rails-ap', '~> 4.1.0', group: :assets
 
 # Keyboard shortcuts
-gem 'mousetrap-rails', '~> 1.4.6'
+gem 'mousetrap-rails', '~> 1.4.6', group: :assets
 
 # Detect and convert string character encoding
 gem 'charlock_holmes', '~> 0.7.3'
@@ -213,22 +213,22 @@
 gem 'chronic', '~> 0.10.2'
 gem 'chronic_duration', '~> 0.10.6'
 
-gem 'sass-rails', '~> 5.0.0'
-gem 'coffee-rails', '~> 4.1.0'
-gem 'uglifier', '~> 2.7.2'
+gem 'sass-rails', '~> 5.0.0', group: :assets
+gem 'coffee-rails', '~> 4.1.0', group: :assets
+gem 'uglifier', '~> 2.7.2', group: :assets
 gem 'turbolinks', '~> 2.5.0'
-gem 'jquery-turbolinks', '~> 2.1.0'
+gem 'jquery-turbolinks', '~> 2.1.0', group: :assets
 
 gem 'addressable',        '~> 2.3.8'
-gem 'bootstrap-sass',     '~> 3.3.0'
+gem 'bootstrap-sass',     '~> 3.3.0', group: :assets
 gem 'font-awesome-rails', '~> 4.6.1'
 gem 'gemojione',          '~> 3.0'
 gem 'gon',                '~> 6.1.0'
-gem 'jquery-atwho-rails', '~> 1.3.2'
-gem 'jquery-rails',       '~> 4.1.0'
-gem 'jquery-ui-rails',    '~> 5.0.0'
+gem 'jquery-atwho-rails', '~> 1.3.2', group: :assets
+gem 'jquery-rails',       '~> 4.1.0', group: :assets
+gem 'jquery-ui-rails',    '~> 5.0.0', group: :assets
 gem 'request_store',      '~> 1.3.0'
-gem 'select2-rails',      '~> 3.5.9'
+gem 'select2-rails',      '~> 3.5.9', group: :assets
 gem 'virtus',             '~> 1.0.1'
 gem 'net-ssh',            '~> 3.0.1'
 gem 'base32',             '~> 0.3.0'
@@ -325,7 +325,7 @@
   gem 'gitlab_meta', '7.0'
 end
 
-gem 'newrelic_rpm', '~> 3.16'
+gem 'newrelic_rpm', '~> 3.16', group: :development
 
 gem 'octokit', '~> 4.3.0'
 
