From: Jakub Jirutka <jakub@jirutka.cz>
Date: Wed, 29 Apr 2016 20:08:37 +0200
Subject: [PATCH] Move some unnecessary gems to groups (i.e. exclude them)

diff --git a/Gemfile b/Gemfile
index 67cc3f3..4ccc74b 100644
--- a/Gemfile
+++ b/Gemfile
@@ -22,19 +22,19 @@ gem 'devise',                 '~> 3.5.4'
 gem 'devise-async',           '~> 0.9.0'
 gem 'doorkeeper',             '~> 2.2.0'
 gem 'omniauth',               '~> 1.3.1'
-gem 'omniauth-auth0',         '~> 1.4.1'
-gem 'omniauth-azure-oauth2',  '~> 0.0.6'
+gem 'omniauth-auth0',         '~> 1.4.1', group: :auth_ext
+gem 'omniauth-azure-oauth2',  '~> 0.0.6', group: :auth_ext
 gem 'omniauth-bitbucket',     '~> 0.0.2'
-gem 'omniauth-cas3',          '~> 1.1.2'
-gem 'omniauth-facebook',      '~> 3.0.0'
+gem 'omniauth-cas3',          '~> 1.1.2', group: :auth_ext
+gem 'omniauth-facebook',      '~> 3.0.0', group: :auth_ext
 gem 'omniauth-github',        '~> 1.1.1'
 gem 'omniauth-gitlab',        '~> 1.0.0'
-gem 'omniauth-google-oauth2', '~> 0.2.0'
+gem 'omniauth-google-oauth2', '~> 0.2.0', group: :auth_ext
 gem 'omniauth-kerberos',      '~> 0.3.0', group: :kerberos
-gem 'omniauth-saml',          '~> 1.5.0'
-gem 'omniauth-shibboleth',    '~> 1.2.0'
-gem 'omniauth-twitter',       '~> 1.2.0'
-gem 'omniauth_crowd',         '~> 2.2.0'
+gem 'omniauth-saml',          '~> 1.5.0', group: :auth_ext
+gem 'omniauth-shibboleth',    '~> 1.2.0', group: :auth_ext
+gem 'omniauth-twitter',       '~> 1.2.0', group: :auth_ext
+gem 'omniauth_crowd',         '~> 2.2.0', group: :auth_ext
 gem 'rack-oauth2',            '~> 1.2.1'
 
 # Spam and anti-bot protection
@@ -81,11 +81,11 @@ gem "haml-rails", '~> 0.9.0'
 gem "carrierwave", '~> 0.10.0'
 
 # Drag and Drop UI
-gem 'dropzonejs-rails', '~> 0.7.1'
+gem 'dropzonejs-rails', '~> 0.7.1', group: :assets
 
 # for aws storage
-gem "fog", "~> 1.36.0"
-gem "unf", '~> 0.1.4'
+gem "fog", "~> 1.36.0", group: :aws
+gem "unf", '~> 0.1.4', group: :aws
 
 # Authorization
 gem "six", '~> 0.2.0'
@@ -175,7 +175,7 @@ gem 'asana', '~> 0.4.0'
 gem 'ruby-fogbugz', '~> 0.2.1'
 
 # d3
-gem 'd3_rails', '~> 3.5.0'
+gem 'd3_rails', '~> 3.5.0', group: :assets
 
 #cal-heatmap
 gem 'cal-heatmap-rails', '~> 3.5.0'
@@ -197,32 +197,32 @@ gem 'licensee', '~> 8.0.0'
 gem "rack-attack", '~> 4.3.1'
 
 # Ace editor
-gem 'ace-rails-ap', '~> 2.0.1'
+gem 'ace-rails-ap', '~> 2.0.1', group: :assets
 
 # Keyboard shortcuts
-gem 'mousetrap-rails', '~> 1.4.6'
+gem 'mousetrap-rails', '~> 1.4.6', group: :assets
 
 # Detect and convert string character encoding
 gem 'charlock_holmes', '~> 0.7.3'
 
-gem "sass-rails", '~> 5.0.0'
-gem "coffee-rails", '~> 4.1.0'
-gem "uglifier", '~> 2.7.2'
+gem "sass-rails", '~> 5.0.0', group: :assets
+gem "coffee-rails", '~> 4.1.0', group: :assets
+gem "uglifier", '~> 2.7.2', group: :assets
 gem 'turbolinks', '~> 2.5.0'
-gem 'jquery-turbolinks', '~> 2.1.0'
+gem 'jquery-turbolinks', '~> 2.1.0', group: :assets
 
 gem 'addressable',        '~> 2.3.8'
-gem 'bootstrap-sass',     '~> 3.3.0'
+gem 'bootstrap-sass',     '~> 3.3.0', group: :assets
 gem 'font-awesome-rails', '~> 4.2'
 gem 'gitlab_emoji',       '~> 0.3.0'
 gem 'gon',                '~> 6.0.1'
-gem 'jquery-atwho-rails', '~> 1.3.2'
-gem 'jquery-rails',       '~> 4.0.0'
-gem 'jquery-scrollto-rails', '~> 1.4.3'
-gem 'jquery-ui-rails',    '~> 5.0.0'
-gem 'raphael-rails',      '~> 2.1.2'
+gem 'jquery-atwho-rails', '~> 1.3.2', group: :assets
+gem 'jquery-rails',       '~> 4.0.0', group: :assets
+gem 'jquery-scrollto-rails', '~> 1.4.3', group: :assets
+gem 'jquery-ui-rails',    '~> 5.0.0', group: :assets
+gem 'raphael-rails',      '~> 2.1.2', group: :assets
 gem 'request_store',      '~> 1.3.0'
-gem 'select2-rails',      '~> 3.5.9'
+gem 'select2-rails',      '~> 3.5.9', group: :assets
 gem 'virtus',             '~> 1.0.1'
 gem 'net-ssh',            '~> 3.0.1'
 
@@ -316,7 +316,7 @@ group :production do
   gem "gitlab_meta", '7.0'
 end
 
-gem "newrelic_rpm", '~> 3.14'
+gem "newrelic_rpm", '~> 3.14', group: :development
 
 gem 'octokit', '~> 4.3.0'
