From: Jakub Jirutka <jakub@jirutka.cz>
Date: Wed, 19 Apr 2016 20:35:00 +0200
Subject: [PATCH] Adjust paths and remove some boilerplate from configs

diff --git a/config/database.yml.postgresql b/config/database.yml.postgresql
index 7067e0f..c4a36e3 100644
--- a/config/database.yml.postgresql
+++ b/config/database.yml.postgresql
@@ -1,45 +1,10 @@
-#
-# PRODUCTION
-#
+---
 production:
   adapter: postgresql
   encoding: unicode
-  database: gitlabhq_production
-  pool: 10
-  # username: git
+  database: gitlab
+  pool: 5
+  # username: gitlab
   # password:
   # host: localhost
-  # port: 5432 
-
-#
-# Development specific
-#
-development:
-  adapter: postgresql
-  encoding: unicode
-  database: gitlabhq_development
-  pool: 5
-  username: postgres
-  password:
-
-#
-# Staging specific
-#
-staging:
-  adapter: postgresql
-  encoding: unicode
-  database: gitlabhq_staging
-  pool: 5
-  username: postgres
-  password:
-
-# Warning: The database defined as "test" will be erased and
-# re-generated from your development database when you run "rake".
-# Do not set this db to the same as development or production.
-test: &test
-  adapter: postgresql
-  encoding: unicode
-  database: gitlabhq_test
-  pool: 5
-  username: postgres
-  password:
+  # port: 5432
diff --git a/config/gitlab.yml.example b/config/gitlab.yml.example
index d9c15f8..f687e2b 100644
--- a/config/gitlab.yml.example
+++ b/config/gitlab.yml.example
@@ -2,15 +2,6 @@
 # GitLab application config file  #
 # # # # # # # # # # # # # # # # # #
 #
-###########################  NOTE  #####################################
-# This file should not receive new settings. All configuration options #
-# * are being moved to ApplicationSetting model!                       #
-# If a setting requires an application restart say so in that screen.  #
-# If you change this file in a Merge Request, please also create       #
-# a MR on https://gitlab.com/gitlab-org/omnibus-gitlab/merge_requests  #
-########################################################################
-#
-#
 # How to use:
 # 1. Copy file as gitlab.yml
 # 2. Update gitlab -> host with your fully qualified domain name
@@ -108,7 +99,7 @@
     ## Repository downloads directory
     # When a user clicks e.g. 'Download zip' on a project, a temporary zip file is created in the following directory.
     # The default is 'shared/cache/archive/' relative to the root of the Rails app.
-    # repository_downloads_path: shared/cache/archive/
+    repository_downloads_path: /var/tmp/gitlab/downloads
 
   ## Reply by email
   # Allow users to comment on issues and merge requests by replying to notification emails.
@@ -145,19 +136,19 @@
   artifacts:
     enabled: true
     # The location where build artifacts are stored (default: shared/artifacts).
-    # path: shared/artifacts
+    path: /var/lib/gitlab/artifacts
 
   ## Git LFS
   lfs:
     enabled: true
     # The location where LFS objects are stored (default: shared/lfs-objects).
-    # storage_path: shared/lfs-objects
+    storage_path: /var/lib/gitlab/lfs-objects
 
   ## GitLab Pages
   pages:
     enabled: false
     # The location where pages are stored (default: shared/pages).
-    # path: shared/pages
+    path: /var/lib/gitlab/pages
 
     # The domain under which the pages are served:
     # http://group.example.com/project
@@ -209,7 +200,7 @@
     # port: 5005
     # api_url: http://localhost:5000/ # internal address to the registry, will be used by GitLab to directly communicate with API
     # key: config/registry.key
-    # path: shared/registry
+    # path: /var/lib/gitlab/registry
     # issuer: gitlab-issuer
 
   #
@@ -226,7 +217,7 @@
     # add_pusher: true
 
     # The location where build traces are stored (default: builds/). Relative paths are relative to Rails.root
-    # builds_path: builds/
+    builds_path: /var/lib/gitlab/builds/
 
   #
   # 3. Auth settings
@@ -271,7 +262,7 @@
         # This setting specifies if LDAP server is Active Directory LDAP server.
         # For non AD servers it skips the AD specific queries.
         # If your LDAP server is not AD, set this to false.
-        active_directory: true
+        active_directory: false
 
         # If allow_username_or_email_login is enabled, GitLab will ignore everything
         # after the first '@' in the LDAP username submitted by the user on login.
@@ -346,7 +337,7 @@
     # This allows users to login without having a user account first. Define the allowed providers
     # using an array, e.g. ["saml", "twitter"], or as true/false to allow all providers or none.
     # User accounts will be created automatically when authentication was successful.
-    allow_single_sign_on: ["saml"]
+    allow_single_sign_on: []
 
     # Locks down those users until they have been cleared by the admin (default: true).
     block_auto_created_users: true
@@ -354,11 +345,6 @@
     # link the omniauth identity with the LDAP account. (default: false)
     auto_link_ldap_user: false
 
-    # Allow users with existing accounts to login and auto link their account via SAML
-    # login, without having to do a manual login first and manually add SAML
-    # (default: false)
-    auto_link_saml_user: false
-
     # Set different Omniauth providers as external so that all users creating accounts
     # via these providers will not be able to have access to internal projects. You
     # will need to use the full name of the provider, like `google_oauth2` for Google.
@@ -374,94 +360,22 @@
     # arguments, followed by optional 'args' which can be either a hash or an array.
     # Documentation for this is available at http://doc.gitlab.com/ce/integration/omniauth.html
     providers:
-      # See omniauth-cas3 for more configuration details
-      # - { name: 'cas3',
-      #     label: 'cas3',
-      #     args: {
-      #             url: 'https://sso.example.com',
-      #             disable_ssl_verification: false,
-      #             login_url: '/cas/login',
-      #             service_validate_url: '/cas/p3/serviceValidate',
-      #             logout_url: '/cas/logout'} }
-      # - { name: 'authentiq',
-      #     # for client credentials (client ID and secret), go to https://www.authentiq.com/
-      #     app_id: 'YOUR_CLIENT_ID',
-      #     app_secret: 'YOUR_CLIENT_SECRET',
-      #     args: {
-      #             scope: 'aq:name email~rs address aq:push'
-      #             # redirect_uri parameter is optional except when 'gitlab.host' in this file is set to 'localhost'
-      #             # redirect_uri: 'YOUR_REDIRECT_URI'
-      #           }
-      #   }
       # - { name: 'github',
       #     app_id: 'YOUR_APP_ID',
       #     app_secret: 'YOUR_APP_SECRET',
       #     url: "https://github.com/",
       #     verify_ssl: true,
       #     args: { scope: 'user:email' } }
-      # - { name: 'bitbucket',
-      #     app_id: 'YOUR_APP_ID',
-      #     app_secret: 'YOUR_APP_SECRET' }
-      # - { name: 'gitlab',
-      #     app_id: 'YOUR_APP_ID',
-      #     app_secret: 'YOUR_APP_SECRET',
-      #     args: { scope: 'api' } }
-      # - { name: 'google_oauth2',
-      #     app_id: 'YOUR_APP_ID',
-      #     app_secret: 'YOUR_APP_SECRET',
-      #     args: { access_type: 'offline', approval_prompt: '' } }
-      # - { name: 'facebook',
-      #     app_id: 'YOUR_APP_ID',
-      #     app_secret: 'YOUR_APP_SECRET' }
-      # - { name: 'twitter',
-      #     app_id: 'YOUR_APP_ID',
-      #     app_secret: 'YOUR_APP_SECRET' }
-      #
-      # - { name: 'saml',
-      #     label: 'Our SAML Provider',
-      #     groups_attribute: 'Groups',
-      #     external_groups: ['Contractors', 'Freelancers'],
-      #     args: {
-      #             assertion_consumer_service_url: 'https://gitlab.example.com/users/auth/saml/callback',
-      #             idp_cert_fingerprint: '43:51:43:a1:b5:fc:8b:b7:0a:3a:a9:b1:0f:66:73:a8',
-      #             idp_sso_target_url: 'https://login.example.com/idp',
-      #             issuer: 'https://gitlab.example.com',
-      #             name_identifier_format: 'urn:oasis:names:tc:SAML:2.0:nameid-format:transient'
-      #           } }
-      #
-      # - { name: 'crowd',
-      #     args: {
-      #       crowd_server_url: 'CROWD SERVER URL',
-      #       application_name: 'YOUR_APP_NAME',
-      #       application_password: 'YOUR_APP_PASSWORD' } }
-      #
-      # - { name: 'auth0',
-      #     args: {
-      #       client_id: 'YOUR_AUTH0_CLIENT_ID',
-      #       client_secret: 'YOUR_AUTH0_CLIENT_SECRET',
-      #       namespace: 'YOUR_AUTH0_DOMAIN' } }
-
-    # SSO maximum session duration in seconds. Defaults to CAS default of 8 hours.
-    # cas3:
-    #   session_duration: 28800
 
   # Shared file storage settings
   shared:
-    # path: /mnt/gitlab # Default: shared
+    path: /var/lib/gitlab  # Default: shared
 
 
   #
   # 4. Advanced settings
   # ==========================
 
-  # GitLab Satellites
-  #
-  # Note for maintainers: keep the satellites.path setting until GitLab 9.0 at
-  # least. This setting is fed to 'rm -rf' in
-  # db/migrate/20151023144219_remove_satellites.rb
-  satellites:
-    path: /home/git/gitlab-satellites/
-
   ## Repositories settings
   repositories:
     # Paths where repositories can be stored. Give the canonicalized absolute pathname.
@@ -469,37 +383,23 @@
     # gitlab-shell invokes Dir.pwd inside the repository path and that results
     # real path not the symlink.
     storages: # You must have at least a `default` storage path.
-      default: /home/git/repositories/
+      default: /var/lib/gitlab/repositories/
 
   ## Backup settings
   backup:
-    path: "tmp/backups"   # Relative paths are relative to Rails.root (default: tmp/backups/)
+    path: /var/tmp/gitlab/backups   # Relative paths are relative to Rails.root (default: tmp/backups/)
     # archive_permissions: 0640 # Permissions for the resulting backup.tar file (default: 0600)
     # keep_time: 604800   # default: 0 (forever) (in seconds)
     # pg_schema: public     # default: nil, it means that all schemas will be backed up
-    # upload:
-    #   # Fog storage connection settings, see http://fog.io/storage/ .
-    #   connection:
-    #     provider: AWS
-    #     region: eu-west-1
-    #     aws_access_key_id: AKIAKIAKI
-    #     aws_secret_access_key: 'secret123'
-    #   # The remote 'directory' to store your backups. For S3, this would be the bucket name.
-    #   remote_directory: 'my.s3.bucket'
-    #   # Use multipart uploads when file size reaches 100MB, see
-    #   #  http://docs.aws.amazon.com/AmazonS3/latest/dev/uploadobjusingmpu.html
-    #   multipart_chunk_size: 104857600
-    #   # Turns on AWS Server-Side Encryption with Amazon S3-Managed Keys for backups, this is optional
-    #   # encryption: 'AES256'
 
   ## GitLab Shell settings
   gitlab_shell:
-    path: /home/git/gitlab-shell/
-    hooks_path: /home/git/gitlab-shell/hooks/
+    path: /usr/share/gitlab-shell/
+    hooks_path: /usr/share/gitlab-shell/hooks/
 
     # File that contains the secret key for verifying access for gitlab-shell.
     # Default is '.gitlab_shell_secret' relative to Rails.root (i.e. root of the GitLab app).
-    # secret_file: /home/git/gitlab/.gitlab_shell_secret
+    secret_file: /etc/gitlab/gitlab-shell.secret
 
     # Git over HTTP
     upload_pack: true
@@ -511,7 +411,7 @@
   workhorse:
     # File that contains the secret key for verifying access for gitlab-workhorse.
     # Default is '.gitlab_workhorse_secret' relative to Rails.root (i.e. root of the GitLab app).
-    # secret_file: /home/git/gitlab/.gitlab_workhorse_secret
+    secret_file: /etc/gitlab/gitlab-workhorse.secret
 
   ## Git settings
   # CAUTION!
@@ -520,16 +420,6 @@
     # Git timeout to read a commit, in seconds
     timeout: 10
 
-  ## Webpack settings
-  # If enabled, this will tell rails to serve frontend assets from the webpack-dev-server running
-  # on a given port instead of serving directly from /assets/webpack. This is only indended for use
-  # in development.
-  webpack:
-    # dev_server:
-    #   enabled: true
-    #   host: localhost
-    #   port: 3808
-
   #
   # 5. Extra customization
   # ==========================
@@ -558,57 +448,3 @@
       #
       # Ban an IP for one hour (3600s) after too many auth attempts
       # bantime: 3600
-
-development:
-  <<: *base
-
-test:
-  <<: *base
-  gravatar:
-    enabled: true
-  lfs:
-    enabled: false
-  gitlab:
-    host: localhost
-    port: 80
-
-    # When you run tests we clone and setup gitlab-shell
-    # In order to setup it correctly you need to specify
-    # your system username you use to run GitLab
-    # user: YOUR_USERNAME
-  satellites:
-    path: tmp/tests/gitlab-satellites/
-  repositories:
-    storages:
-      default: tmp/tests/repositories/
-  backup:
-    path: tmp/tests/backups
-  gitlab_shell:
-    path: tmp/tests/gitlab-shell/
-    hooks_path: tmp/tests/gitlab-shell/hooks/
-  issues_tracker:
-    redmine:
-      title: "Redmine"
-      project_url: "http://redmine/projects/:issues_tracker_id"
-      issues_url: "http://redmine/:project_id/:issues_tracker_id/:id"
-      new_issue_url: "http://redmine/projects/:issues_tracker_id/issues/new"
-    jira:
-      title: "JIRA"
-      url: https://sample_company.atlasian.net
-      project_key: PROJECT
-  ldap:
-    enabled: false
-    servers:
-      main:
-        label: ldap
-        host: 127.0.0.1
-        port: 3890
-        uid: 'uid'
-        method: 'plain' # "tls" or "ssl" or "plain"
-        base: 'dc=example,dc=com'
-        user_filter: ''
-        group_base: 'ou=groups,dc=example,dc=com'
-        admin_group: ''
-
-staging:
-  <<: *base
\ No newline at end of file
--- a/config/initializers/smtp_settings.rb.sample
+++ b/config/initializers/smtp_settings.rb.sample
@@ -1,16 +1,9 @@
-# To enable smtp email delivery for your GitLab instance do the following:
-# 1. Rename this file to smtp_settings.rb
-# 2. Edit settings inside this file
-# 3. Restart GitLab instance
-#
 # For full list of options and their values see http://api.rubyonrails.org/classes/ActionMailer/Base.html
-#
-# If you change this file in a Merge Request, please also create a Merge Request on https://gitlab.com/gitlab-org/omnibus-gitlab/merge_requests
 
 if Rails.env.production?
-  Rails.application.config.action_mailer.delivery_method = :smtp
+  #Rails.application.config.action_mailer.delivery_method = :smtp
+  #ActionMailer::Base.delivery_method = :smtp
 
-  ActionMailer::Base.delivery_method = :smtp
   ActionMailer::Base.smtp_settings = {
     address: "email.server.com",
     port: 465,
--- a/config/resque.yml.example
+++ b/config/resque.yml.example
@@ -1,20 +1,6 @@
-# If you change this file in a Merge Request, please also create
-# a Merge Request on https://gitlab.com/gitlab-org/omnibus-gitlab/merge_requests
-#
-development:
-  url: redis://localhost:6379
-  # sentinels:
-  #   -
-  #     host: localhost
-  #     port: 26380 # point to sentinel, not to redis port
-  #   -
-  #     host: slave2
-  #     port: 26381 # point to sentinel, not to redis port
-test:
-  url: redis://localhost:6379
 production:
   # Redis (single instance)
-  url: unix:/var/run/redis/redis.sock
+  url: unix:/run/redis/redis.sock
   ##
   # Redis + Sentinel (for HA)
   #
--- a/config/sidekiq.yml.example
+++ b/config/sidekiq.yml.example
@@ -1,2 +1,2 @@
 ---
-:concurrency: 5
\ No newline at end of file
+:concurrency: 25
--- a/config/unicorn.rb.example
+++ b/config/unicorn.rb.example
@@ -22,24 +22,19 @@
 # Read about unicorn workers here:
 # http://doc.gitlab.com/ee/install/requirements.html#unicorn-workers
 #
-worker_processes 3
-
-# Since Unicorn is never exposed to outside clients, it does not need to
-# run on the standard HTTP port (80), there is no reason to start Unicorn
-# as root unless it's from system init scripts.
-# If running the master process as root and the workers as an unprivileged
-# user, do this to switch euid/egid in the workers (also chowns logs):
-# user "unprivileged_user", "unprivileged_group"
+worker_processes ENV.fetch('UNICORN_WORKERS', 2).to_i
 
 # Help ensure your application will always spawn in the symlinked
 # "current" directory that Capistrano sets up.
-working_directory "/home/git/gitlab" # available in 0.94.0+
+working_directory ENV.fetch('GITLAB_BASE', Dir.pwd) # available in 0.94.0+
 
 # Listen on both a Unix domain socket and a TCP port.
 # If you are load-balancing multiple Unicorn masters, lower the backlog
 # setting to e.g. 64 for faster failover.
-listen "/home/git/gitlab/tmp/sockets/gitlab.socket", :backlog => 1024
-listen "127.0.0.1:8080", :tcp_nopush => true
+listen ENV['UNICORN_LISTEN_TCP'], :tcp_nopush => true
+if ENV['UNICORN_LISTEN_UNIX']
+  listen ENV['UNICORN_LISTEN_UNIX'], :backlog => 1024
+end
 
 # nuke workers after 30 seconds instead of 60 seconds (the default)
 #
@@ -56,16 +51,7 @@ listen "127.0.0.1:8080", :tcp_nopush => true
 #
 # For more information see http://stackoverflow.com/a/21682112/752049
 #
-timeout 60
-
-# feel free to point this anywhere accessible on the filesystem
-pid "/home/git/gitlab/tmp/pids/unicorn.pid"
-
-# By default, the Unicorn logger will write to stderr.
-# Additionally, some applications/frameworks log to stderr or stdout,
-# so prevent them from going to /dev/null when daemonized here:
-stderr_path "/home/git/gitlab/log/unicorn.stderr.log"
-stdout_path "/home/git/gitlab/log/unicorn.stdout.log"
+timeout ENV.fetch('UNICORN_TIMEOUT', 60).to_i
 
 # combine Ruby 2.0.0dev or REE with "preload_app true" for memory savings
 # http://rubyenterpriseedition.com/faq.html#adapt_apps_for_cow
@@ -87,23 +73,6 @@ before_fork do |server, worker|
   defined?(ActiveRecord::Base) and
     ActiveRecord::Base.connection.disconnect!
 
-  # The following is only recommended for memory/DB-constrained
-  # installations.  It is not needed if your system can house
-  # twice as many worker_processes as you have configured.
-  #
-  # This allows a new master process to incrementally
-  # phase out the old master process with SIGTTOU to avoid a
-  # thundering herd (especially in the "preload_app false" case)
-  # when doing a transparent upgrade.  The last worker spawned
-  # will then kill off the old master process with a SIGQUIT.
-  old_pid = "#{server.config[:pid]}.oldbin"
-  if old_pid != server.pid
-    begin
-      sig = (worker.nr + 1) >= server.worker_processes ? :QUIT : :TTOU
-      Process.kill(sig, File.read(old_pid).to_i)
-    rescue Errno::ENOENT, Errno::ESRCH
-    end
-  end
   #
   # Throttle the master from forking too quickly by sleeping.  Due
   # to the implementation of standard Unix signal handlers, this
