From: Jakub Jirutka <jakub@jirutka.cz>
Date: Wed, 28 Apr 2016 02:36:00 +0200
Subject: [PATCH] Move .secret to config/secret_token

diff --git a/config/initializers/secret_token.rb b/config/initializers/secret_token.rb
index dae3a4a..472a232 100644
--- a/config/initializers/secret_token.rb
+++ b/config/initializers/secret_token.rb
@@ -8,7 +8,7 @@ require 'securerandom'
 # no regular words or you'll be exposed to dictionary attacks.
 
 def find_secure_token
-  token_file = Rails.root.join('.secret')
+  token_file = "#{Rails.root}/config/secret_token"
   if ENV.key?('SECRET_KEY_BASE')
     ENV['SECRET_KEY_BASE']
   elsif File.exist? token_file
