From: Jakub Jirutka <jakub@jirutka.cz>
Date: Wed, 29 Apr 2016 00:03:00 +0200
Subject: [PATCH] Fix gitlab:setup task to not drop & create database

diff --git a/lib/tasks/gitlab/setup.rake b/lib/tasks/gitlab/setup.rake
index 48baecf..df74626 100644
--- a/lib/tasks/gitlab/setup.rake
+++ b/lib/tasks/gitlab/setup.rake
@@ -14,8 +14,8 @@ namespace :gitlab do
       puts ""
     end
 
-    Rake::Task["db:reset"].invoke
-    Rake::Task["add_limits_mysql"].invoke
+    Rake::Task["db:schema:load"].invoke
+    Rake::Task["db:seed"].invoke
     Rake::Task["setup_postgresql"].invoke
     Rake::Task["db:seed_fu"].invoke
   rescue Gitlab::TaskAbortedByUserError
