From d6e4518650ce743a647442fcc471b489bd52089a Mon Sep 17 00:00:00 2001
From: Jakub Jirutka <jakub@jirutka.cz>
Date: Thu, 20 Jul 2017 17:43:00 +0200
Subject: [PATCH 10/10] Enable Asciidoctor's attribute "sectanchors" by default

---
 lib/gitlab/asciidoc.rb | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/gitlab/asciidoc.rb b/lib/gitlab/asciidoc.rb
index 3d41ac7..2a84322 100644
--- a/lib/gitlab/asciidoc.rb
+++ b/lib/gitlab/asciidoc.rb
@@ -8,7 +8,8 @@ module Gitlab
   module Asciidoc
     DEFAULT_ADOC_ATTRS = [
       'showtitle', 'idprefix=user-content-', 'idseparator=-', 'env=gitlab',
-      'env-gitlab', 'source-highlighter=html-pipeline', 'icons=font'
+      'env-gitlab', 'source-highlighter=html-pipeline', 'icons=font',
+      'sectanchors'
     ].freeze
 
     # Public: Converts the provided Asciidoc markup into HTML.
-- 
2.10.1 (Apple Git-78)

