From b15c998fae135f92c71d2707616a6f8a0ea540bc Mon Sep 17 00:00:00 2001
From: Jakub Jirutka <jakub@jirutka.cz>
Date: Wed, 28 Feb 2018 18:03:29 +0100
Subject: [PATCH] Expose some project metadata as AsciiDoc attributes

---
 lib/gitlab/asciidoc.rb | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

diff --git a/lib/gitlab/asciidoc.rb b/lib/gitlab/asciidoc.rb
index 81e2583..ce84709 100644
--- a/lib/gitlab/asciidoc.rb
+++ b/lib/gitlab/asciidoc.rb
@@ -40,9 +40,25 @@ module Gitlab
         )
       end
 
+      attributes = if (project = context[:project])
+        {
+          'gl-commit-ref' => context[:ref] || context[:project].default_branch,
+          'gl-project-namespace' => project.namespace.full_path,
+          'gl-project-path' => project.full_path,
+          'gl-project-title' => project.title,
+          'gl-project-url' => project.web_url,
+          'gl-repo-http-url' => project.http_url_to_repo,
+          'gl-repo-ssh-url' => project.ssh_url_to_repo,
+          'gl-server-host' => Gitlab.config.gitlab.host,
+          'gl-server-url' => Gitlab.config.gitlab.base_url,
+        }.map { |k, v| "#{k}=#{v}" }
+      else
+        []
+      end
+
       asciidoc_opts = { safe: :server,
                         backend: :gitlab_html5,
-                        attributes: DEFAULT_ADOC_ATTRS,
+                        attributes: DEFAULT_ADOC_ATTRS + attributes,
                         extensions: extensions }
 
       context[:pipeline] = :ascii_doc
--
2.10.1 (Apple Git-78)

