From: Jakub Jirutka <jakub@jirutka.cz>
Date: Fri, 31 Sep 2017 2:31:00 +0200
Subject: [PATCH] Make location of gitlab_workhorse_secret file configurable

--- a/config/gitlab.yml.example
+++ b/config/gitlab.yml.example
@@ -508,6 +508,11 @@
     # If you use non-standard ssh port you need to specify it
     # ssh_port: 22
 
+  workhorse:
+    # File that contains the secret key for verifying access for gitlab-workhorse.
+    # Default is '.gitlab_workhorse_secret' relative to Rails.root (i.e. root of the GitLab app).
+    # secret_file: /home/git/gitlab/.gitlab_workhorse_secret
+
   ## Git settings
   # CAUTION!
   # Use the default values unless you really know what you are doing
--- a/config/initializers/1_settings.rb
+++ b/config/initializers/1_settings.rb
@@ -357,6 +357,12 @@
 Settings.gitlab_shell['ssh_path_prefix'] ||= Settings.send(:build_gitlab_shell_ssh_path_prefix)
 
 #
+# Workhorse
+#
+Settings['workhorse'] ||= Settingslogic.new({})
+Settings.workhorse['secret_file'] ||= Rails.root.join('.gitlab_workhorse_secret')
+
+#
 # Repositories
 #
 Settings['repositories'] ||= Settingslogic.new({})
--- a/lib/gitlab/workhorse.rb
+++ b/lib/gitlab/workhorse.rb
@@ -151,7 +151,7 @@
       end
 
       def secret_path
-        Rails.root.join('.gitlab_workhorse_secret')
+        Gitlab.config.workhorse.secret_file
       end
 
       protected
