From b2464aabf37637189549db9cd2fb05b2c87c73d0 Mon Sep 17 00:00:00 2001
From: Jakub Jirutka <jakub@jirutka.cz>
Date: Tue, 20 Sep 2016 23:51:00 +0200
Subject: [PATCH 05/10] Disable checking init script in gitlab:check task

---
 lib/tasks/gitlab/check.rake | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/lib/tasks/gitlab/check.rake b/lib/tasks/gitlab/check.rake
index 858f1cd..bab6b0d 100644
--- a/lib/tasks/gitlab/check.rake
+++ b/lib/tasks/gitlab/check.rake
@@ -27,8 +27,8 @@ namespace :gitlab do
         SystemCheck::App::UploadsDirectoryExistsCheck,
         SystemCheck::App::UploadsPathPermissionCheck,
         SystemCheck::App::UploadsPathTmpPermissionCheck,
-        SystemCheck::App::InitScriptExistsCheck,
-        SystemCheck::App::InitScriptUpToDateCheck,
+        #SystemCheck::App::InitScriptExistsCheck,
+        #SystemCheck::App::InitScriptUpToDateCheck,
         SystemCheck::App::ProjectsHaveNamespaceCheck,
         SystemCheck::App::RedisVersionCheck,
         SystemCheck::App::RubyVersionCheck,
@@ -272,13 +272,6 @@ namespace :gitlab do
         puts "yes".color(:green)
       else
         puts "no".color(:red)
-        try_fixing_it(
-          sudo_gitlab("RAILS_ENV=production bin/background_jobs start")
-        )
-        for_more_information(
-          see_installation_guide_section("Install Init Script"),
-          "see log/sidekiq.log for possible errors"
-        )
         fix_and_rerun
       end
     end
-- 
2.10.1 (Apple Git-78)

