--- a/Makefile
+++ b/Makefile
@@ -175,6 +177,7 @@
 build: executors/docker/bindata.go $(GOX)
 	# Building $(NAME) in version $(VERSION) for $(BUILD_PLATFORMS)
 	gox $(BUILD_PLATFORMS) \
+		-verbose \
 		-ldflags "$(GO_LDFLAGS)" \
 		-output="out/binaries/$(NAME)-{{.OS}}-{{.Arch}}" \
 		$(PKG)
@@ -182,7 +185,7 @@
 build_simple: $(GOPATH_SETUP)
 	# Building $(NAME) in version $(VERSION) for current platform
 	go build \
-		-ldflags "$(GO_LDFLAGS)" \
+		-v -ldflags "$(GO_LDFLAGS)" \
 		-o "out/binaries/$(NAME)" \
 		$(PKG)
 
