--- a/gitaly-proto.gemspec
+++ b/gitaly-proto.gemspec
@@ -15,7 +15,7 @@
   spec.homepage      = "https://gitlab.com/gitlab-org/gitaly-proto"
   spec.license       = "MIT"
 
-  spec.files         = `git ls-files -z #{prefix}`.split("\x0").reject { |f| f.match(%r{^#{prefix}/(test|spec|features)/}) }
+  spec.files         = Dir["#{prefix}/lib/**/*"]
   spec.require_paths = ["#{prefix}/lib"]
 
   spec.add_dependency "google-protobuf", "~> 3.1"
