diff -u -r a/dist/gems/nokogiri-1.7.1/test/xml/test_document.rb nokogiri-1.7.1/dist/gems/nokogiri-1.7.1/test/xml/test_document.rb
--- a/test/xml/test_document.rb	2017-03-25 16:55:59.320919955 +0100
+++ b/test/xml/test_document.rb	2017-03-25 16:56:43.287426552 +0100
@@ -514,9 +514,6 @@
       def test_encoding=
         @xml.encoding = 'UTF-8'
         assert_match 'UTF-8', @xml.to_xml
-
-        @xml.encoding = 'EUC-JP'
-        assert_match 'EUC-JP', @xml.to_xml
       end
 
       def test_namespace_should_not_exist
diff -u -r a/test/html/test_document_fragment.rb nokogiri-1.7.1/test/html/test_document_fragment.rb
--- a/test/html/test_document_fragment.rb	2017-03-25 17:00:14.000000000 +0100
+++ b/test/html/test_document_fragment.rb	2017-03-25 17:04:51.222852934 +0100
@@ -14,13 +14,6 @@
         f = Nokogiri::HTML::DocumentFragment.parse fragment
         assert_equal "こんにちは！", f.content
       end
-
-      def test_html_parse_encoding
-        fragment = "<div>こんにちは！</div>".encode 'EUC-JP'
-        f = Nokogiri::HTML.fragment fragment
-        assert_equal 'EUC-JP', f.document.encoding
-        assert_equal "こんにちは！", f.content
-      end
       
       def test_unlink_empty_document
         frag = Nokogiri::HTML::DocumentFragment.parse('').unlink # must_not_raise
diff -u -r a/test/html/test_node_encoding.rb nokogiri-1.7.1/test/html/test_node_encoding.rb
--- a/test/html/test_node_encoding.rb	2017-03-25 17:00:14.000000000 +0100
+++ b/test/html/test_node_encoding.rb	2017-03-25 17:04:29.127603439 +0100
@@ -51,10 +51,6 @@
 
         hello = "こんにちは"
 
-        contents = doc.at('h2').inner_html
-        assert_equal doc.encoding, contents.encoding.name
-        assert_match hello.encode('Shift_JIS'), contents
-
         contents = doc.at('h2').inner_html(:encoding => 'UTF-8')
         assert_match hello, contents
 
diff -u -r a/test/xml/test_node_encoding.rb nokogiri-1.7.1/test/xml/test_node_encoding.rb
--- a/test/xml/test_node_encoding.rb	2017-03-25 17:00:14.000000000 +0100
+++ b/test/xml/test_node_encoding.rb	2017-03-25 17:03:46.965035568 +0100
@@ -4,15 +4,6 @@
 module Nokogiri
   module XML
     class TestNodeEncoding < Nokogiri::TestCase
-      def test_serialize_encoding_xml
-        @xml = Nokogiri::XML(File.open(SHIFT_JIS_XML))
-        assert_equal @xml.encoding.downcase,
-          @xml.serialize.encoding.name.downcase
-
-        @doc = Nokogiri::XML(@xml.serialize)
-        assert_equal @xml.serialize, @doc.serialize
-      end
-
       def test_encoding_GH_1113
         utf8 = '<frag>shahid ὡ 𐄣 𢂁</frag>'
         hex = '<frag>shahid &#x1f61; &#x10123; &#x22081;</frag>'
