--- a/src/ngx_http_lua_module.c
+++ b/src/ngx_http_lua_module.c
@@ -771,7 +771,7 @@
         dd("initializing lua vm");
 
 #ifndef OPENRESTY_LUAJIT
-        if (ngx_process != NGX_PROCESS_SIGNALLER && !ngx_test_config) {
+        if (ngx_process != NGX_PROCESS_SIGNALLER && !ngx_test_config && !getenv("NGX_LUA_IGNORE_LUAJIT")) {
             ngx_log_error(NGX_LOG_ALERT, cf->log, 0,
                           "detected a LuaJIT version which is not OpenResty's"
                           "; many optimizations will be disabled and "
