--- a/SuiteSparse_config/SuiteSparse_config.mk	2016-06-04 21:50:58.511999012 +0200
+++ b/SuiteSparse_config/SuiteSparse_config.mk	2016-06-04 21:51:11.155575272 +0200
@@ -338,7 +338,7 @@
 
     ifeq ($(UNAME),Linux)
         # add the realtime library, librt, and SuiteSparse/lib
-        LDLIBS += -lrt -Wl,-rpath=$(INSTALL_LIB)
+        LDLIBS += -lrt
     endif
 
     #---------------------------------------------------------------------------
