--- a/logging.go
+++ b/logging.go
@@ -42,6 +42,7 @@
 type logConfiguration struct {
 	logFile   string
 	logFormat string
+	disableAccessLog bool
 }
 
 func startLogging(config logConfiguration) {
@@ -73,7 +74,9 @@
 		log.WithField("logFormat", config.logFormat).Fatal("Unknown logFormat configured")
 	}
 
-	helper.SetAccessLoggerEntry(accessLogEntry)
+	if !config.disableAccessLog {
+		helper.SetAccessLoggerEntry(accessLogEntry)
+	}
 	log.SetOutput(logOutputWriter)
 
 	// Golog always goes to stderr
--- a/main.go
+++ b/main.go
@@ -62,6 +62,7 @@
 func init() {
 	flag.StringVar(&logConfig.logFile, "logFile", "", "Log file location")
 	flag.StringVar(&logConfig.logFormat, "logFormat", "text", "Log format to use defaults to text (text, json, structured, none)")
+	flag.BoolVar(&logConfig.disableAccessLog, "disableAccessLog", false, "Disable access log")
 }
 
 func main() {
