--- a/logging.go
+++ b/logging.go
@@ -61,9 +61,11 @@
 	case noneLogType:
 		accessLogEntry = nil
 		logOutputWriter = ioutil.Discard
+		goLog.SetOutput(os.Stderr)
 	case jsonLogFormat:
 		accessLogEntry = log.WithField("system", "http")
 		log.SetFormatter(&log.JSONFormatter{})
+		goLog.SetOutput(os.Stderr)
 	case textLogFormat:
 		accessLogger := log.New()
 		accessLogger.Formatter = helper.NewAccessLogFormatter()
@@ -72,10 +74,12 @@
 		accessLogEntry = accessLogger.WithField("system", "http")
 
 		log.SetFormatter(&log.TextFormatter{})
+		goLog.SetOutput(logOutputWriter)
 	case structuredFormat:
 		formatter := &log.TextFormatter{ForceColors: true, EnvironmentOverrideColors: true}
 		log.SetFormatter(formatter)
 		accessLogEntry = log.WithField("system", "http")
+		goLog.SetOutput(logOutputWriter)
 	default:
 		log.WithField("logFormat", config.logFormat).Fatal("Unknown logFormat configured")
 	}
@@ -84,8 +88,4 @@
 		helper.SetAccessLoggerEntry(accessLogEntry)
 	}
 	log.SetOutput(logOutputWriter)
-
-	// Golog always goes to stderr
-	goLog.SetOutput(os.Stderr)
-
 }
