--- a/redis.gemspec
+++ b/redis.gemspec
@@ -35,9 +35,7 @@
 
   s.email = ["redis-db@googlegroups.com"]
 
-  s.files         = `git ls-files`.split("\n")
-  s.test_files    = `git ls-files -- {test,spec,features}/*`.split("\n")
-  s.executables   = `git ls-files -- bin/*`.split("\n").map{ |f| File.basename(f) }
+  s.files         = Dir["lib/**/*"]
 
   s.add_development_dependency("rake", "<11.0.0")
   s.add_development_dependency("test-unit", "3.1.5")
