--- a/ext/digest/digest_conf.rb
+++ b/ext/digest/digest_conf.rb
@@ -65,18 +65,14 @@

     dir_config("openssl")
     pkg_config("openssl")
-    require File.expand_path('../../openssl/deprecation', __FILE__)
     if find_openssl_library
       funcs ||= name.upcase
       funcs = Array(funcs)
       types ||= funcs
       hdr = "openssl/#{hdr}.h"
-      if funcs.all? {|func| OpenSSL.check_func("#{func}_Transform", hdr)} &&
-         types.all? {|type| have_type("#{type}_CTX", hdr)}
-        $defs << "-D#{name.upcase}_USE_OPENSSL"
-        $headers << "#{name}ossl.h"
-        return :ossl
-      end
+      $defs << "-D#{name.upcase}_USE_OPENSSL"
+      $headers << "#{name}ossl.h"
+      return :ossl
     end
   end
   $objs << "#{name}.#{$OBJEXT}"
