--- a/config.toml.example
+++ b/config.toml.example
@@ -1,9 +1,9 @@
 # Example Gitaly configuration file
 
-socket_path = "/home/git/gitlab/tmp/sockets/private/gitaly.socket"
+socket_path = "/run/gitlab/gitaly.socket"
 
 # The directory where Gitaly's executables are stored
-bin_dir = "/home/git/gitaly"
+bin_dir = "/usr/bin"
 
 # # Optional: listen on a TCP socket. This is insecure (no authentication)
 # listen_addr = "localhost:9999"
@@ -22,7 +22,7 @@
 
 [[storage]]
 name = "default"
-path = "/home/git/repositories"
+path = "/var/lib/gitlab/repositories"
 
 # # You can optionally configure more storages for this Gitaly instance to serve up
 #
@@ -31,13 +31,13 @@
 # path = "/mnt/other_storage/repositories"
 #
 
-# # You can optionally configure Gitaly to output JSON-formatted log messages to stdout
-# [logging]
+# You can optionally configure Gitaly to output JSON-formatted log messages to stdout
+[logging]
 # format = "json"
-# # Optional: Set log level to only log entries with that severity or above
-# # One of, in order: debug, info, warn, errror, fatal, panic
-# # Defaults to "info"
-# level = "warn"
+# Optional: Set log level to only log entries with that severity or above
+# One of, in order: debug, info, warn, errror, fatal, panic
+# Defaults to "info"
+level = "warn"
 #
 # # Additionally exceptions from the Go server can be reported to Sentry
 # sentry_dsn = "https://<key>:<secret>@sentry.io/<project>"
@@ -50,7 +50,7 @@
 
 [gitaly-ruby]
 # The directory where gitaly-ruby is installed
-dir = "/home/git/gitaly/ruby"
+dir = "/usr/lib/bundles/gitaly-ruby"
 
 # # Gitaly-ruby resident set size (RSS) that triggers a memory restart (bytes)
 # max_rss = 200000000
@@ -66,7 +66,7 @@
 
 [gitlab-shell]
 # The directory where gitlab-shell is installed
-dir = "/home/git/gitlab-shell"
+dir = "/usr/share/gitlab-shell"
 
 # # You can adjust the concurrency of each RPC endpoint
 # [[concurrency]]
