See comment in packed_binaries.go.

--- a/Makefile
+++ b/Makefile
@@ -327,7 +327,7 @@
 ## Install Gitaly binaries. The target directory can be modified by setting PREFIX and DESTDIR.
 install: build
 	${Q}mkdir -p ${INSTALL_DEST_DIR}
-	install ${GITALY_EXECUTABLES} "${INSTALL_DEST_DIR}"
+	install $(filter-out ${GITALY_PACKED_EXECUTABLES},${GITALY_EXECUTABLES}) "${INSTALL_DEST_DIR}"
 
 .PHONY: build-bundled-git
 ## Build bundled Git binaries.
