--- a/hc/settings.py
+++ b/hc/settings.py
@@ -6,9 +6,11 @@
 """
 
 import os
+import sys
 import warnings
 
-BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+HC_SETTINGS = os.environ.get('HC_SETTINGS', '/etc/healthchecks/settings.py')
+HC_MODULE_DIR = os.path.dirname(os.path.abspath(__file__))
 
 
 def envbool(s, default):
@@ -72,7 +74,7 @@
 TEMPLATES = [
     {
         'BACKEND': 'django.template.backends.django.DjangoTemplates',
-        'DIRS': [os.path.join(BASE_DIR, 'templates')],
+        'DIRS': [os.path.join(HC_MODULE_DIR, 'templates')],
         'APP_DIRS': True,
         'OPTIONS': {
             'context_processors': [
@@ -95,7 +97,7 @@
 DATABASES = {
     'default': {
         'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': os.getenv("DB_NAME", BASE_DIR + "/hc.sqlite"),
+        'NAME': os.getenv("DB_NAME", "/var/lib/healthchecks/hc.sqlite"),
     }
 }
 
@@ -146,8 +148,8 @@
 PING_ENDPOINT = os.getenv("PING_ENDPOINT", SITE_ROOT + "/ping/")
 PING_EMAIL_DOMAIN = os.getenv("PING_EMAIL_DOMAIN", "localhost")
 STATIC_URL = '/static/'
-STATICFILES_DIRS = [os.path.join(BASE_DIR, "static")]
-STATIC_ROOT = os.path.join(BASE_DIR, 'static-collected')
+STATICFILES_DIRS = [os.path.join(HC_MODULE_DIR, 'static')]
+STATIC_ROOT = os.path.join(sys.prefix, 'static')
 STATICFILES_FINDERS = (
     'django.contrib.staticfiles.finders.FileSystemFinder',
     'django.contrib.staticfiles.finders.AppDirectoriesFinder',
@@ -201,8 +203,3 @@
 MATRIX_HOMESERVER = os.getenv("MATRIX_HOMESERVER")
 MATRIX_USER_ID = os.getenv("MATRIX_USER_ID")
 MATRIX_ACCESS_TOKEN = os.getenv("MATRIX_ACCESS_TOKEN")
-
-if os.path.exists(os.path.join(BASE_DIR, "hc/local_settings.py")):
-    from .local_settings import *
-else:
-    warnings.warn("local_settings.py not found, using defaults")
