--- a/hc/settings.py
+++ b/hc/settings.py
@@ -11,9 +11,11 @@
 """

 import os
+import sys
 import warnings

-BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+HC_SETTINGS = os.environ.get('HC_SETTINGS', '/etc/healthchecks/settings.py')
+HC_MODULE_DIR = os.path.dirname(os.path.abspath(__file__))

 HOST = "localhost"
 SECRET_KEY = "---"
@@ -61,7 +64,7 @@
 TEMPLATES = [
     {
         'BACKEND': 'django.template.backends.django.DjangoTemplates',
-        'DIRS': [os.path.join(BASE_DIR, 'templates')],
+        'DIRS': [os.path.join(HC_MODULE_DIR, 'templates')],
         'APP_DIRS': True,
         'OPTIONS': {
             'context_processors': [
@@ -84,7 +87,7 @@
 DATABASES = {
     'default': {
         'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': '{0}/hc.sqlite'.format(BASE_DIR),
+        'NAME': '/var/lib/healthchecks/hc.sqlite',
     }
 }

@@ -125,8 +128,8 @@
 PING_ENDPOINT = SITE_ROOT + "/ping/"
 PING_EMAIL_DOMAIN = HOST
 STATIC_URL = '/static/'
-STATICFILES_DIRS = [os.path.join(BASE_DIR, "static")]
-STATIC_ROOT = os.path.join(BASE_DIR, 'static-collected')
+STATICFILES_DIRS = [os.path.join(HC_MODULE_DIR, 'static')]
+STATIC_ROOT = os.path.join(sys.prefix, 'static')
 STATICFILES_FINDERS = (
     'django.contrib.staticfiles.finders.FileSystemFinder',
     'django.contrib.staticfiles.finders.AppDirectoriesFinder',
@@ -168,8 +171,3 @@
 # Zendesk
 ZENDESK_CLIENT_ID = None
 ZENDESK_CLIENT_SECRET = None
-
-if os.path.exists(os.path.join(BASE_DIR, "hc/local_settings.py")):
-    from .local_settings import *
-else:
-    warnings.warn("local_settings.py not found, using defaults")
